File: resolve_merge.rb
files = `git diff --name-only --diff-filter=U`.split("\n")
files.each do |file|
  content = File.read(file)
  if content.include?("<<<<<<< HEAD")
    # We will keep HEAD, but if we see dashboard_controller, we manually merge
    if file == "app/controllers/dashboard_controller.rb"
      content = content.gsub(/<<<<<<< HEAD\n(.*?)\n=======\n(.*?)\n>>>>>>> default\/develop/m, "\\1,\n\\2")
      File.write(file, content)
    elsif file == "app/controllers/api/v1/accounts/inboxes_controller.rb"
      # Merge both blocks
      content = content.gsub(/<<<<<<< HEAD\n(.*?)\n=======\n(.*?)\n>>>>>>> default\/develop/m, "\\1\n\\2")
      File.write(file, content)
    elsif file == "app/services/whatsapp/oneoff_campaign_service.rb"
      # We just manually fixed this in our previous thought, wait we didn't, the multi replace failed.
      # Let's keep HEAD for now, and I will replace it manually right after.
      system("git checkout --ours '#{file}'")
    else
      # For all others, we keep our changes (HEAD) because they contain the custom Evolution/Kanban features
      system("git checkout --ours '#{file}'")
    end
    system("git add '#{file}'")
  end
end

File: resolve_merge.py
import os
import subprocess

files = subprocess.check_output(['git', 'diff', '--name-only', '--diff-filter=U']).decode('utf-8').split('\n')
for file in files:
    if not file:
        continue
    with open(file, 'r') as f:
        content = f.read()
    
    if "<<<<<<< HEAD" in content:
        if file == "app/controllers/dashboard_controller.rb":
            import re
            content = re.sub(r'<<<<<<< HEAD\n(.*?)\n=======\n(.*?)\n>>>>>>> default/develop', r'\1,\n\2', content, flags=re.DOTALL)
            with open(file, 'w') as f:
                f.write(content)
            subprocess.run(['git', 'add', file])
        elif file == "app/controllers/api/v1/accounts/inboxes_controller.rb":
            import re
            content = re.sub(r'<<<<<<< HEAD\n(.*?)\n=======\n(.*?)\n>>>>>>> default/develop', r'\1\n\2', content, flags=re.DOTALL)
            with open(file, 'w') as f:
                f.write(content)
            subprocess.run(['git', 'add', file])
        else:
            subprocess.run(['git', 'checkout', '--ours', file])
            subprocess.run(['git', 'add', file])
